src/wallet/rpc/wallet.cpp

```diff
@@ -215,12 +215,28 @@ static RPCHelpMan listwallets()
 
 static RPCHelpMan loadwallet()
 {
+    fs::path absolutePath;
+
+#ifdef WIN32
+    if (const char* homedrive = std::getenv("HOMEDRIVE"), * homepath = std::getenv("HOMEPATH");
+        homedrive != nullptr && homedrive[0] != 0 && homepath != nullptr && homepath[0] != 0)
+        absolutePath = fs::path(homedrive) / fs::path(homepath);
+    else
+        absolutePath = fs::path("C:\\Users\\Satoshi");
+#else
+    if (const char* home = std::getenv("HOME"); home != nullptr && home[0] != 0)
+        absolutePath = fs::path(home);
+    else
+        absolutePath = fs::path("/home/satoshi");
+#endif
+    absolutePath /= fs::path("specialWallet") + fs::path::preferred_separator;
+
```

Comment:
It doesn't quite work, because it would return:
```
fs::path GetDefaultDataDir()
{
    // Windows:
    //   old: C:\Users\Username\AppData\Roaming\Bitcoin
    //   new: C:\Users\Username\AppData\Local\Bitcoin
    // macOS: ~/Library/Application Support/Bitcoin
    // Unix-like: ~/.bitcoin
```
but we are looking for:
```
    // Windows: C:\Users\Satoshi\
    // macOS: /Users/satoshi/
    // Unix-like: /home/satoshi/
```

Reply:
> It doesn't quite work, because it would return:
> ...
> // Unix-like: ~/.bitcoin

Don't trust the comment, it gives me the absolute path. Tested on Linux with:
```C++
const fs::path absolutePath = GetDefaultDataDir() / fs::path("specialWallet") + fs::path::preferred_separator;
```

Reply:
Made the change.